### PR TITLE
Added `databricks labs ucx validate-groups-membership` command to validate groups to see if they have same membership across acount and workspace level

### LINKED ISSUE #649: Catch groups that doesn't have the same membership on Workspace and Account level
In groups.py, when we find a corresponding group in the Account console we do not check if the number of Users is the same.
We should log the Account groups that have the same name as a Workspace group, but a different membership

### PR DESCRIPTION
**Pull request by prajin-29**
This pull request introduces several changes related to managing Databricks workspace groups. Here's a summary of the changes:

1. A new command `validate-groups-membership` is added to the CLI tool in the file `cli.py`. This command validates the groups in the workspace and account levels to see if they have different memberships. It returns a table showing the differences between the groups.
2. A new class `GroupManager` is added to the file `groups.py` to manage the groups in the workspace and account levels. This class includes a new method `validate_group_membership` that compares the members of each group in the workspace and account levels and returns a list of groups that have different memberships.
3. A new fixture `make_ucx_group_with_diff_members` is added to the file `conftest.py` to create a pair of groups with different members.
4. A new test case `test_group_matching_names_with_diff_users` is added to the file `test_groups.py` to test the `validate_group_membership` method with a pair of groups that have different members.
5. A retries decorator is added to the test case `test_replace_workspace_groups_with_account_groups` to handle potential intermittent failures.

These changes improve the group management functionality of the tool and make it more robust and reliable. The new `validate-groups-membership` command provides valuable information about the groups in the workspace and account levels, and the new `GroupManager` class and test cases ensure that the group management functionality works as expected.

Closes #649

### CODE REVIEW

File: src/databricks/labs/ucx/workspace_access/groups.py [hunk 1]
```diff
@@ -38,6 +38,7 @@ class MigratedGroup:
     entitlements: str | None = None
     external_id: str | None = None
     roles: str | None = None
+    is_same_membership: bool | None = None
```
Comment (nfx) [inline src/databricks/labs/ucx/workspace_access/groups.py:41]: no, you're not allowed to add this column, as it'll be a breaking change to database, which this small feature is not justifying

**Review by nfx — CHANGES_REQUESTED** [review]
do this functionality as a CLI command. You can rely on `GroupManager`, but don't rely on `ucx.groups` table - e.g. it should be possible to run `databricks labs ucx validate-groups` before running the assessment workflow

File: src/databricks/labs/ucx/workspace_access/groups.py (hunk 1 above)
Comment (prajin-29) [reply src/databricks/labs/ucx/workspace_access/groups.py]: Changed the logic to incorporate it with CLI.

**Comment by prajin-29** [comment]
> do this functionality as a CLI command. You can rely on `GroupManager`, but don't rely on `ucx.groups` table - e.g. it should be possible to run `databricks labs ucx validate-groups` before running the assessment workflow

@nfx .Did this in CLI Command. Can you please validate the same.

File: tests/integration/conftest.py [hunk 2]
```diff
@@ -105,6 +105,26 @@ def inner(workspace_group_name=None, account_group_name=None):
     return inner
 
 
+@pytest.fixture
+def make_ucx_group_with_diff_members(make_random, make_group, make_acc_group, make_user):
```
Comment (nfx) [inline tests/integration/conftest.py:109]: Don't add a fixture if it's used only once. It makes the code convoluted and harder to maintain. Rewrite by inlining the fixture code into that single test

File: tests/unit/workspace_access/test_groups.py [hunk 3]
```diff
@@ -824,3 +824,62 @@ def test_state():
     assert not state.get_target_principal("invalid_group_name")
     assert not state.get_temp_principal("invalid_group_name")
     assert not state.is_in_scope("invalid_group_name")
+
+
+def test_validate_group_diff_membership():
+    backend = MockBackend()
+    wsclient = MagicMock()
```
Comment (nfx) [inline tests/unit/workspace_access/test_groups.py:831]: Use "create_autospec(WorkspaceClient)"

File: tests/unit/workspace_access/test_groups.py [hunk 4]
```diff
@@ -824,3 +824,62 @@ def test_state():
...
+    }
+    grp_membership = GroupManager(
+        backend, wsclient, inventory_database="inv", workspace_group_regex=r"\(([1-9]+)\)", account_group_regex="[1-9]+"
+    ).validate_group_membership()
+    assert grp_membership == [{"wf_group_name": "test_(1234)", "ac_group_name": "ac_test_1234"}]
+
+
+def test_validate_group_same_membership():
+    backend = MockBackend()
+    wsclient = MagicMock()
```
Comment (nfx) [inline tests/unit/workspace_access/test_groups.py:858]: Create autospec

**Review by nfx — CHANGES_REQUESTED** [review]

File: src/databricks/labs/ucx/cli.py [hunk 5]
```diff
@@ -138,6 +139,37 @@ def repair_run(w: WorkspaceClient, step):
...
+        ws=w,
+        inventory_database=inventory_database,
+        include_group_names=include_group_names,
+        renamed_group_prefix=renamed_group_prefix,
+        workspace_group_regex=workspace_group_regex,
+        workspace_group_replace=workspace_group_replace,
+        account_group_regex=account_group_regex,
+    )
+    mismatch_groups = group_manager.validate_group_membership()
+    print(json.dumps(mismatch_groups))
```
Comment (william-conti) [inline src/databricks/labs/ucx/cli.py:170]: It should be better to return an error code when there's at least one entry in the mismatch_groups

File: src/databricks/labs/ucx/cli.py [hunk 6]
```diff
@@ -138,6 +139,37 @@ def repair_run(w: WorkspaceClient, step):
     installer.repair_run(step)
 
 
+@ucx.command
+def validate_groups_membership(w: WorkspaceClient):
```
Comment (william-conti) [inline src/databricks/labs/ucx/cli.py:143]: What if I wan't to scope it for all workspaces in my account ?

**Review by william-conti — CHANGES_REQUESTED** [review]
We should plan on enlarging this feature for all the workspaces in the account

File: src/databricks/labs/ucx/cli.py (hunk 6 above)
Comment (nfx) [reply src/databricks/labs/ucx/cli.py:143]: It has to be an additional command

File: labs.yml [hunk 7]
```diff
@@ -80,3 +80,10 @@ commands:
         description: target catalog to migrate schema to
       - name: to-schema
         description: target schema to migrate tables to
+
+  - name: validate-groups-membership
+    description: Validate the groups to see if the groups at account level and workspace level have different membership
+    table_template: |-
+      Workflow_Group_Name\tAccount_Group_Name
```
Comment (nfx) [inline labs.yml:87]: Use spaces for separating words, not underscores

File: labs.yml [hunk 8]
```diff
@@ -80,3 +80,10 @@ commands:
...
+      {{range .}}{{.wf_group_name}}\t{{.ac_group_name}}
```
Comment (nfx) [inline labs.yml:88]:
```suggestion
      {{range .}}{{.name_in_workspace}}\t{{.name_in_account}}
```
Use conventional variable names

File: src/databricks/labs/ucx/cli.py (hunk 5 above)
Comment (nfx) [reply src/databricks/labs/ucx/cli.py:170]: I'm not sure if CLI wrapper supports that just yet

**Review by nfx — CHANGES_REQUESTED** [review]
1. Inline fixture that is not reused
2. Make variable names consistent 
3. Add additional account level command

File: tests/integration/conftest.py (hunk 2 above)
Comment (prajin-29) [reply tests/integration/conftest.py]: Removed this fixture and rewrote it inline with test.

File: tests/unit/workspace_access/test_groups.py (hunk 3 above)
Comment (prajin-29) [reply tests/unit/workspace_access/test_groups.py]: Used create_autospec

File: labs.yml (hunk 7 above)
Comment (prajin-29) [reply labs.yml]: Replaced _ with space

File: labs.yml (hunk 8 above)
Comment (prajin-29) [reply labs.yml]: Renamed with the conventional names

**Review by nfx — APPROVED** [review]

**Pull request merged**